lib/types/inet-address.js

```diff
@@ -37,42 +37,55 @@ InetAddress.fromString = function (value) {
   if (!value) {
     return new InetAddress(new Buffer([0, 0, 0, 0]));
   }
-  var pattern = /^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$|^[\da-f:]+$/i;
-  if (!pattern.test(value)) {
-    throw new TypeError('Value could not be parsed as InetAddress: ' + value);
-  }
-  var parts = value.split('.');
-  if (parts.length === 4) {
-    //IPv4
+  var ipv4Pattern = /^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$/;
+  var ipv6Pattern = /^[\da-f:.]+$/i;
```

Comment:
I'm not sure if we should support interface suffix, or at least not in this patch.
About the ipv6 regex, it will also match ipv4 to handle embedded IPv4 addresses in IPv6 strings, but IPv4 addressed will be handled by `ipv4Pattern.test()` below.